### DISCUSSION TITLE
Clipping a surface with a polygon

### DISCUSSION THREAD

**Discussion by ThibaultFaney** [pyvista/pyvista#1696] [Q&A] [answered]
Hi,

I have a surface mesh created from a set of points using a Delaunay_2D triangulation : 

cloud = pv.PolyData(points)   #  points is a set of points with x,y,z coordinates
surf = cloud.delaunay_2d()

I also have a polygon corresponding to a subset of the 2D area covered by the triangulation (either as a shapely 2D polygon or a set of points with x,y coordinates).

Is it possible to clip  the surface mesh so that its boundaries become the polygon boundaries ? Essentially, I want to mesh the inside of the polygon, without losing the information of the z coordinate in the original surface mesh.

**Comment by akaszynski** [comment, marked as answer]
Surface clipping using VTK's boolean operations seem to only work with 3D surfaces as it relies on normals.

One alternative might be to add both of the polygons together and then proceed to use [delaunay_2d](https://docs.pyvista.org/api/core/_autosummary/pyvista.PolyData.delaunay_2d.html)

```py
import pyvista
pyvista.set_plot_theme('document')

bg_circle = pyvista.Circle(radius=1)
sm_circle = pyvista.Circle(radius=0.75)

merged = bg_circle + sm_circle
merged.delaunay_2d(alpha=0.5).plot()
```
![tmp](https://user-images.githubusercontent.com/11981631/134958722-10f27b2a-5fd0-408a-a923-c14b6a621521.png)
